etcdctl/ctlv3/command/make_mirror_command.go

```diff
@@ -60,23 +67,61 @@ func NewMakeMirrorCommand() *cobra.Command {
 	return c
 }
 
+func destinationEndpoints() ([]string, error) {
+	discoveryCfg := &discoveryCfg{
+		domain:      mmdiscoverySrv,
+		serviceName: mmdiscoverySrvName,
+	}
+
+	if discoveryCfg.domain != "" {
+		srvs, err := srv.GetClient("etcd-client", discoveryCfg.domain, discoveryCfg.serviceName)
+		if err != nil {
+			return nil, err
+		}
+		return srvs.Endpoints, nil
+	}
+	return mmendpoints, nil
+}
+
 func makeMirrorCommandFunc(cmd *cobra.Command, args []string) {
-	if len(args) != 1 {
-		ExitWithError(ExitBadArgs, errors.New("make-mirror takes one destination argument"))
+	if len(args) == 0 && len(mmendpoints) == 0 && mmdiscoverySrv == "" {
+		ExitWithError(ExitBadArgs, errors.New("No destination endpoint(s) specified, --dest-endpoints or --dest-discovery-srv required"))
+	}
+	if len(mmendpoints) > 0 && mmdiscoverySrv != "" {
+		ExitWithError(ExitBadArgs, errors.New("--dest-endpoints and --dest-discovery-srv are mutually exclusive"))
+	}
+	if len(args) >= 1 && mmdiscoverySrv != "" {
+		ExitWithError(ExitBadArgs, errors.New("destination argument and --dest-discovery-srv are mutually exclusive"))
+	}
+	if len(args) == 1 {
+		if len(mmendpoints) != 0 {
+			ExitWithError(ExitBadArgs, errors.New("Multiple destination endpoints must be specified using --dest-endpoints"))
+		}
+		// For backwards compatibility accept single destination endpoint specified as first positional argument
+		mmendpoints = []string{args[0]}
+	}
+	if len(args) > 1 {
+		ExitWithError(ExitBadArgs, errors.New("make-mirror takes one destination argument. Multiple destination endpoints must be specified with --dest-endpoints"))
```

Comment:
Keeping backward compatibility opens up a lot of incompatible flag and arg permutations. I'm pretty sure I've covered them all here and given decent errors messages.